chain/beacon/drand/drand.go

```diff
@@ -239,13 +250,41 @@ var _ beacon.RandomBeacon = (*DrandBeacon)(nil)
 
 func BeaconScheduleFromDrandSchedule(dcs dtypes.DrandSchedule, genesisTime uint64, ps *pubsub.PubSub) (beacon.Schedule, error) {
 	shd := beacon.Schedule{}
-	for _, dc := range dcs {
+	for i, dc := range dcs {
 		bc, err := NewDrandBeacon(genesisTime, buildconstants.BlockDelaySecs, ps, dc.Config)
 		if err != nil {
-			return nil, xerrors.Errorf("creating drand beacon: %w", err)
+			return nil, xerrors.Errorf("%d creating drand beacon: %w", i, err)
 		}
 		shd = append(shd, beacon.BeaconPoint{Start: dc.Start, Beacon: bc})
 	}
 
 	return shd, nil
 }
+
+var _ dclient.Client = historicalBeaconClient{}
+
+// historicalBeaconClient is a drand client that doesn't actually do anything. It's used when
+// we don't have a drand network to connect to but still need to provide a beacon client.
+// We don't expect calls through to the client to be made since we should only be verifying old
+// randomness, not fetching it.
+type historicalBeaconClient struct{}
+
+func (h historicalBeaconClient) Get(ctx context.Context, round uint64) (dclient.Result, error) {
+	return nil, xerrors.Errorf("no historical randomness available")
+}
+
+func (h historicalBeaconClient) Watch(ctx context.Context) <-chan dclient.Result {
+	return nil
+}
+
+func (h historicalBeaconClient) Info(ctx context.Context) (*dchain.Info, error) {
+	return nil, xerrors.Errorf("no historical randomness available")
+}
+
+func (h historicalBeaconClient) RoundAt(time.Time) uint64 {
+	return 0
+}
+
+func (h historicalBeaconClient) Close() error {
+	return nil
+}
```

Comment:
I'm very surprised the `Empty` client isn't working for your needs?

Reply:
ahh, TIL about `EmptyClientWithInfo`, but yes, we still have the log problem with it, see below